cirq-core/cirq/sim/simulator_base.py

```diff
@@ -86,22 +90,47 @@ def __init__(
         noise: 'cirq.NOISE_MODEL_LIKE' = None,
         seed: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None,
         ignore_measurement_results: bool = False,
+        split_untangled_states: bool = False,
     ):
         """Initializes the simulator.
 
         Args:
-           dtype: The `numpy.dtype` used by the simulation.
-           noise: A noise model to apply while simulating.
-           seed: The random seed to use for this simulator.
-           ignore_measurement_results: If True, then the simulation
-               will treat measurement as dephasing instead of collapsing
-               process. This is only applicable to simulators that can
-               model dephasing.
+            dtype: The `numpy.dtype` used by the simulation.
+            noise: A noise model to apply while simulating.
+            seed: The random seed to use for this simulator.
+            ignore_measurement_results: If True, then the simulation
+                will treat measurement as dephasing instead of collapsing
+                process. This is only applicable to simulators that can
+                model dephasing.
+            split_untangled_states: If True, optimizes simulation by running
+                unentangled qubit sets independently and merging those states
+                at the end.
         """
         self._dtype = dtype
         self._prng = value.parse_random_state(seed)
         self.noise = devices.NoiseModel.from_noise_model_like(noise)
         self._ignore_measurement_results = ignore_measurement_results
+        self._split_untangled_states = split_untangled_states
+
+    @abc.abstractmethod
+    def _create_act_on_arg(
```

Comment:
Nit: while this name makes sense in comparison with `_create_act_on_args`, using the singular here doesn't work on its own. The `Args` in `ActOnArgs` refers to qubits, prng, measure results, etc.; by extension, `_create_act_on_arg` would create one of those items, rather than a state on a subset of qubits.

`_create_partial_act_on_args` may be clearer.

Reply:
I think what I'd prefer is to rename the existing one to `_create_operation_target` and then have this one be `_create_act_on_args`. That would break backwards compatibility though. :( I'll go with `_create_partial...`.